sunxspex/photon_power_law.py

```diff
@@ -0,0 +1,204 @@
+import functools
+import numpy as np
+import astropy.units as u
+
+PHOTON_RATE_UNIT = u.ph / u.keV / u.cm**2 / u.s
+
+
+@u.quantity_input
+def integrate_power_law(
+    energy: u.keV,
+    norm_energy: u.keV,
+    norm: PHOTON_RATE_UNIT,
+    index: u.one
+) -> (PHOTON_RATE_UNIT * u.keV):
+    r"""Evaluate the antiderivative of a power law at a given energy or vector of energies.
+
+    The power law antiderivative evaluated by this function is assumed to take the following form,
+    :math:`f(E) = N \left( \frac{E}{E_0} \right)^{- \gamma}`,
+    where :math:`E` is the energy, :math:`N` is the normalization, :math:`E_0` is the normalization energy,
+    and :math:`\gamma` is the power law index.
+
+    The value of :math:`\gamma` is assumed to be positive, but the functional form
+    includes a negative sign.
+
+    The special case of :math:`\gamma = 1` is handled.
+
+    Parameters
+    ----------
+    energy : `astropy.units.Quantity`
+        Energy (or vector of energies) at which to evaluate the power law antiderivative.
+    norm_energy : `astropy.units.Quantity`
+        Energy used for the normalization of the power law argument, i.e. :math:`E_0`.
+    norm : `astropy.units.Quantity`
+        Normalization of the power law integral, i.e. :math:`N`, in units convertible to ph / (cm2 . keV . s).
+    index : `astropy.units.Quantity`
+        The power law index, i.e. :math:`\gamma`.
+
+    Returns
+    -------
+    `astropy.units.Quantity`
+        Analytical antiderivative of a power law evaluated at the given energies.
+    """
+    prefactor = norm * norm_energy
+    arg = (energy / norm_energy).to(u.one)
+    if index == 1:
+        return prefactor * np.log(arg)
+    return (prefactor / (1 - index)) * arg**(1 - index)
+
+
+@u.quantity_input
+def compute_broken_power_law(
+    energy_edges: u.keV,
+    reference_energy: u.keV,
+    reference_flux: PHOTON_RATE_UNIT,
```

Comment:
if there a reason this is called "reference_flux" rather that "normalization"?

Reply:
i think it's an artifact from when the function had a slightly different implementation. i'll update the name!